test/distribution/uniform_unittest.cpp

```diff
@@ -0,0 +1,52 @@
+#include <autoppl/distribution/uniform.hpp>
+#include <autoppl/expression/variable.hpp>
+
+#include <cmath>
+#include <array>
+
+#include "gtest/gtest.h"
+
+namespace ppl {
+
+struct uniform_dist_fixture : ::testing::Test {
+protected:
+    Variable<double> x {0.5};
+    Variable<double> y {0.1};
+    Uniform<double, double> dist1 = Uniform(0., 1.);
+    Uniform<double, Variable<double> > dist2 = Uniform(0., x);
+    Uniform<Variable<double>, Variable<double> > dist3 = Uniform(y, x);
+};
+
+TEST_F(uniform_dist_fixture, sanity_uniform_test) {
+    EXPECT_EQ(dist1.min(), 0.0);
+    EXPECT_EQ(dist1.max(), 1.0);
+
+    EXPECT_EQ(dist2.min(), 0.0);
+    EXPECT_EQ(dist2.max(), 0.5);
+
+    EXPECT_EQ(dist3.min(), 0.1);
+    EXPECT_EQ(dist3.max(), 0.5);
+}
+
+TEST_F(uniform_dist_fixture, simple_uniform) {
+    EXPECT_DOUBLE_EQ(dist1.pdf(1.1), 0.0);
+    EXPECT_DOUBLE_EQ(dist1.pdf(1.0), 0.0);
+
+    EXPECT_DOUBLE_EQ(dist2.pdf(0.25), 2.0);
+    EXPECT_DOUBLE_EQ(dist2.pdf(-0.1), 0.0);
+
+    EXPECT_DOUBLE_EQ(dist3.pdf(0.25), 2.5);
+}
+
+TEST_F(uniform_dist_fixture, uniform_sampling) {
```

Comment:
I've been thinking about how to test samples - this may be something fun to try. Let's keep this for now since it is a good sanity check. I wonder if it's possible to check more stuff tho. Maybe take a large noeugh sample, compute the sample mean and show that it's near the true mean? What do you think?

Reply:
Possible. There's always some chance of failure, so it's risky. But we might be able to bound the probability of failure over n trials to show that test failures are incredibly unlikely.